src/pages/promocodes/edit-promocode-page.js

```diff
@@ -68,6 +68,10 @@ class EditPromocodePage extends React.Component {
         const title = (entity.id) ? T.translate("general.edit") : T.translate("general.add");
         const breadcrumb = (entity.id) ? entity.code : T.translate("general.new");
 
+        if (entity.id && match.params.promocode_id != entity.id) {
```

Comment:
this hide the issue
u should be doing this on constructor as u do in componentDidUpdate

 const promoCodeId = props.match.params.promocode_id;
        super(props);

        if (!promoCodeId) {
            props.resetPromocodeForm();
        } else {
            props.getPromocode(promoCodeId);
        }